app/usage.go

```diff
@@ -5,11 +5,22 @@ package app
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/services/cache"
 	"github.com/mattermost/mattermost-server/v6/utils"
 )
 
+var usagesCache = cache.NewLRU(cache.LRUOptions{
+	Size:          1,
+	DefaultExpiry: time.Hour,
+})
+
+const (
+	UsageStorageKey = "storage"
+)
+
```

Comment:
@agnivade This PR is currently WIP while I add tests and some polish. I wanted to get some early feedback.

Originally the idea was to calculate the storage usage on every file upload (through the api/app), but turns out plugins often just directly update the `fileInfo` table (e.g. https://github.com/mattermost/focalboard/pull/3052/files#r874707796)

I'm going to work on getting all Mattermost plugins to use the api/app, but right now we're missing some Plugin API methods that plugins could use. There could also be plugins which we don't control that could directly update the FileInfo table. 

This PR can't wait on those changes, so decided I to use a cache here (credit to @vish9812  for the idea)
I'm not sure this is a permanent change, so I didn't opt to make the extensive localcachelayer changes. 

What do you think?